apps/astarte_realm_management/lib/astarte_realm_management/engine.ex

```diff
@@ -458,7 +458,8 @@ defmodule Astarte.RealmManagement.Engine do
          simple_trigger_maps = build_simple_trigger_maps(serialized_tagged_simple_triggers),
          trigger = build_trigger(trigger_name, simple_trigger_maps, action),
          %Trigger{trigger_uuid: trigger_uuid} = trigger,
-         target = build_trigger_target_container("trigger_engine", trigger_uuid),
+         {exchange, routing_key, opts} = target_from_action(action),
+         target = build_trigger_target_container(exchange, routing_key, trigger_uuid, opts),
```

Comment:
Idea: since we already have the `AMQPTriggerTarget` struct, you could build that in `target_from_action` (passing also `trigger_uuid` instead of returning a tuple, then you pass the `AMQPTriggerTarget in `build_trigger_target_container`